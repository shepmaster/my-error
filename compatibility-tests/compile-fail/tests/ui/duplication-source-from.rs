use snafu::Snafu;

#[derive(Debug, Snafu)]
enum EnumError {
    AVariant {
        // First source, legitimate
        source: String,

        // source(from) implies source, so should be a duplicate
        #[snafu(source(from(EnumError, Box::new)))]
        source2: Box<EnumError>,
    },
}

fn main() {}
