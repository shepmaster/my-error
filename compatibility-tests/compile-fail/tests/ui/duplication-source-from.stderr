error: Multiple `source` attributes are not supported within an enum variant
  --> $DIR/duplication-source-from.rs:10:17
   |
10 |         #[snafu(source(from(EnumError, Box::new)))]
   |                 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
